recipes/cpu_features/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CpuFeaturesConan(ConanFile):
+    name = "cpu_features"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/cpu_features"
+    description = "A cross platform C99 library to get cpu features at runtime."
+    topics = ("conan", "cpu", "features", "cpuid")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "_build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["BUILD_PIC"] = self.options.get_safe("fPIC", False)
+        cmake.configure()
+        self._cmake = cmake
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["cpu_features"]
```

Comment:
Not the expert but I am pretty sure this is it

https://github.com/google/cpu_features/blob/7ed0b0e50e9b999c0a95c4e15e746585fe0056da/CMakeLists.txt#L238

https://github.com/google/cpu_features/blob/7ed0b0e50e9b999c0a95c4e15e746585fe0056da/CMakeLists.txt#L244


```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "CpuFeatures"
        self.cpp_info.names["cmake_find_package_multi"] = "CpuFeatures"
        self.cpp_info.components["libcpu_features"].names["cmake_find_package"] = "cpu_features"
        self.cpp_info.components["libcpu_features"].names["cmake_find_package_multi"] = "cpu_features"
        self.cpp_info.components["libcpu_features"].libs = ["cpu_features"]
```

Reply:
not sure, do I need components if there is only one component?

Reply:
Yes because the filename and namespace are different from the default that Conan uses... There's two different strings
- CpuFeatures
- cpu_features

Reply:
so, it couldn't be done by a single component?

Reply:
There is only one component 😕 

There's the global names and the component names that are being configured

Reply:
yeah, I meant default, or global component, if it's possible to name it that way :)
anyway, seems like one is needed, so let's add it

Reply:
It would be great to have a detailed explanation of how the components, namespaces and filenames work and interact with a specifiс complex example. Official docs ("Define the package information") are vague and too general. A CCI wiki page could be a good idea.

Reply:
https://github.com/conan-io/conan-center-index/pull/5080